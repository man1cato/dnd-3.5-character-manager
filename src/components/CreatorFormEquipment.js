import React, { useState, useEffect, Fragment } from 'react'
import _ from 'lodash'
import update from 'immutability-helper'
import { ErrorMessage } from 'formik'

import ItemModal from './ItemModal'
import Selector from './Selector'
import EquipButton from './EquipButton'


const CreatorFormEquipment = ({
	values, 
	items, 
	setFieldValue, 
	setFieldTouched
}) => {
	const [selectedItemIds, setSelectedItemIds] = useState([])
	const [selected, setSelected] = useState(undefined)
	const { equipment, equipped } = values

	useEffect(() => {
		if (selectedItemIds.length > equipment.length) {
			setFieldValue('equipment', [...equipment, { id: _.last(selectedItemIds), qty: 1 }])
		} else {
			const idToRemove = _.difference(_.map(equipment, (item) => item.id), selectedItemIds)[0]
			setFieldValue('equipment', _.filter(equipment, (item) => item.id !== idToRemove))
			setFieldValue('equipped', {
				armor: equipped.armor === idToRemove ? null : equipped.armor,
				shield: equipped.shield === idToRemove ? null : equipped.shield,
				weapons: _.filter(equipped.weapons, (id) => id !== idToRemove)
			})
		}
	}, [selectedItemIds])

	const handleEquip = (e) => {
		const id = e.target.id
		const category = e.target.name
		const alreadyEquipped = _.includes(equipped[category], id) 
		if(category === 'weapons') {
			const weapons = alreadyEquipped ? _.without(equipped.weapons, id) : [...equipped.weapons, id]
			return setFieldValue('equipped', update(equipped, {
				weapons: { $set: weapons }
			}))
		} 
		return setFieldValue('equipped', update(equipped, {
			[category]: { $set: alreadyEquipped ? null : id }
		}))			
	}

	const handleQtyChange = (e) => {
		const id = e.target.id
		const value = e.target.value
		const index = _.findIndex(equipment, {id})
		setFieldValue(`equipment[${index}]`, { id, qty: value })
	}



	const Content = ({selected}) => (
		<div>
			<div className="form-group--35"><b>Category:</b> <div>{selected.category}</div></div>

			{selected.weaponType && (
				<>
					<div className="form-group--35"><b>Type:</b> <div>{selected.weaponType} / {selected.encumbrance} / {selected.damageType}</div></div>
					<div className="form-group--35"><b>Range:</b> <div>{selected.range}</div></div>
					<div className="form-group--35"><b>Damage (M):</b> <div>{selected.damageM}</div></div>
					<div className="form-group--35"><b>Damage (S):</b> <div>{selected.damageS}</div></div>
					<div className="form-group--35"><b>Critical:</b> <div>{selected.critical}</div></div>
				</>
			)}

			<div className="form-group--35"><b>Value:</b> <div>{selected.value} gp</div></div>
			<div className="form-group--35"><b>Weight:</b> <div>{selected.weight} lbs</div></div>
		</div>
	)
	
	return (
		<>
			<h3 className="row--center">Select Equipment</h3>

			<Selector
				apiObject={items} 
				fieldName="equipment"
				Content={Content}
				selectedObjIds={selectedItemIds}
				setSelectedObjIds={setSelectedItemIds}
			/>

			<div className="divider"></div>

			<h4 className="row--left">Selected Equipment:</h4>
			<div className="form-grid--equipment">
				<h5 className="grid__col1">Item</h5>
				<h5 className="grid__col2">Qty</h5>

				{_.map(equipment, (item) => {
					const id = item.id
					item = { id, ...items[id] }
					return (
						<Fragment key={id}>
							<button
								className="grid__col1 button--link"
								type="button"
								id={id}
								onClick={() => setSelected(item)}
							>
								{items[id].name}
							</button>

							<input 
								className="grid__col2 number-input"
								id={id}
								value={_.find(equipment, { id }).qty} 
								onChange={(e) => handleQtyChange(e)} 
							/>
							
							<EquipButton
								className="grid__col3"
								item={item} 
								equipped={equipped} 
								handleEquip={handleEquip} 
							/>
							
							<button 
								className="grid__col4 button"
								type="button" 
								onClick={() => setSelectedItemIds(_.without(selectedItemIds, id))}
							>
								x
							</button>
						</Fragment>
					)
				})}
			</div>

			<ItemModal
				selected={selected}
				equipped={equipped}
				handleCloseModal={() => setSelected(undefined)}
				handleEquip={handleEquip}
			/>

			<ErrorMessage className="row--left form-group--error" name="equipment" component="div" />
		</>
	)
}

export default CreatorFormEquipment