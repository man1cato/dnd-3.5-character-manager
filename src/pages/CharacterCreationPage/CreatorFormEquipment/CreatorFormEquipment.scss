@import '../CharacterCreationPage.scss';


.CreatorFormEquipment__grid {
   @extend .CreatorForm__grid;
   grid-template-columns: 4fr repeat(3, 2fr) 1fr;
   input {
      max-width: 3rem;
   }
}