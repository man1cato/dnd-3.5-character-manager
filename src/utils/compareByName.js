export default (a,b) => {
    const aName = a.name.toUpperCase();
    const bName = b.name.toUpperCase();
    if (aName < bName) { return -1 };
    if (aName > bName) { return 1 };
    return 0;
}