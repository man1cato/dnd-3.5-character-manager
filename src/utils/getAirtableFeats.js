import dotenv from 'dotenv'
dotenv.config()

import axios from 'axios'
import _ from 'lodash'

const apiKey = process.env.AIRTABLE_API_KEY
const baseUrl = 'https://api.airtable.com/v0/appK7TZeddGqjGUDL'

export default async () => {
	try {
		let offset
		let feats = []

		do {
			const res = await axios.get(`${baseUrl}/Feats?api_key=${apiKey}` + (!!offset && `&offset=${offset}` || '') )
			offset = res.data.offset
			const featGroup = res.data.records.map((feat) =>  ({
				id: feat.id,
				name: feat.fields.Name,
				type: feat.fields.Type.join(", ") || null,
				description: feat.fields.Description || null
			}));
			
			feats = feats.concat(featGroup)
		} while (!!offset)
		
		return _.orderBy(feats, ['name'], ['asc'])
	} catch (e) {
		console.error(e)
	}
}