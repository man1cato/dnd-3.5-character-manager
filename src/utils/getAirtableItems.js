import dotenv from 'dotenv'
dotenv.config()

import axios from 'axios'
import _ from 'lodash'

const apiKey = process.env.AIRTABLE_API_KEY
const baseUrl = 'https://api.airtable.com/v0/appK7TZeddGqjGUDL'

export default async () => {
	try {
		let offset;
		let items = [];
		do {
			const res = await axios.get(`${baseUrl}/Items?api_key=${apiKey}` + (!!offset && `&offset=${offset}` || ''))
			offset = res.data.offset
			const itemGroup = res.data.records.map((item) =>  ({
				id: item.id,
				name: item.fields.Name,
				value: item.fields["Value (gp)"] || '',
				weight: item.fields["Weight (lbs)"] || '',
				category: item.fields.Category || null,
				weaponType: item.fields["Weapon Type"] || null,
				encumbrance: item.fields.Encumbrance || null,
				damageType: item.fields["Damage Type"] ? item.fields["Damage Type"].join(" / ") : [],
				damageS: item.fields["Damage (S)"] || null,
				damageM: item.fields["Damage (M)"] || null,
				critical: item.fields.Critical || null,
				range: item.fields.Range || null,
				armorBonus: item.fields["Armor Bonus"] || null,
				maxDexBonus: item.fields["Max DEX Bonus"] || null
			}))
			
			items = items.concat(itemGroup)
		} while (!!offset)
		
		return _.orderBy(items, ['name'], ['asc'])
	} catch (e) {
		console.error(e)
	}
}