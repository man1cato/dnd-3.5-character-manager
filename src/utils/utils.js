import _ from 'lodash'


export const convertTextToArray = (string) => _.compact(string.split(/\n/))

export const findItemById = (storeItemArray, itemId) => storeItemArray.find((item) => item.id === itemId)

export const apiObjectToArray = (obj) => _.orderBy(Object.keys(obj).map((id) => ({
	id,
	...obj[id]
})), ['name'], ['asc'])