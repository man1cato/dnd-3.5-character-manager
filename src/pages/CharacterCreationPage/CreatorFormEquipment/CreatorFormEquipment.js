import React, { useState, useEffect, Fragment } from 'react'
import _ from 'lodash'
import update from 'immutability-helper'
import { ErrorMessage } from 'formik'

import ItemModal from '../../../components/Modals/ItemModal'
import Selector from '../../../components/Selector/Selector'
import EquipButton from '../../../components/EquipButton/EquipButton'
import { apiObjectToArray, calcItemTotalValue, calcEquipmentTotalValue, calcEquipmentTotalWeight } from '../../../utils/utils'
import './CreatorFormEquipment.scss'


const CreatorFormEquipment = ({
	values, 
	items, 
	setFieldValue
}) => {
	const { race, equipped } = values
	const itemsArray = apiObjectToArray(items)

	const [equipment, setEquipment] = useState(values.equipment)
	const [availableItems, setAvailableItems] = useState(itemsArray)
	const [selectedItemIds, setSelectedItemIds] = useState([])
	const [clickedItem, setClickedItem] = useState(undefined)
	const [totalCost, setTotalCost] = useState(0)
	const [totalWeight, setTotalWeight] = useState(0)

	const updateEquipment = equipment => {
		setFieldValue('equipment', equipment)
		setEquipment(equipment)
	}

	useEffect(() => {
		setAvailableItems(_.filter(itemsArray, item =>
			!_.includes(selectedItemIds, item.id)
		))
		if (selectedItemIds.length > equipment.length) {
			updateEquipment([...equipment, { id: _.last(selectedItemIds), qty: 1 }])
		} else {
			const idToRemove = _.difference(_.map(equipment, item => item.id), selectedItemIds)[0]
			updateEquipment(_.filter(equipment, item => item.id !== idToRemove))
			setFieldValue('equipped', {
				armor: equipped.armor === idToRemove ? null : equipped.armor,
				shield: equipped.shield === idToRemove ? null : equipped.shield,
				weapons: _.filter(equipped.weapons, id => id !== idToRemove)
			})
		}
	}, [selectedItemIds])
	
	useEffect(() => {
		setTotalCost(calcEquipmentTotalValue(equipment, items))
		setTotalWeight(calcEquipmentTotalWeight(equipment, items))
	}, [equipment])

	useEffect(() => {
		setFieldValue('remainingGold', (values.startingGold - totalCost).toFixed(2))
	}, [totalCost])

	const handleSelect = selectedItemId => {
		setSelectedItemIds([...selectedItemIds, selectedItemId])
	}

	const handleEquip = (e) => {
		const id = e.target.id
		const category = e.target.name
		const alreadyEquipped = _.includes(equipped[category], id) 
		if(category === 'weapons') {
			const weapons = alreadyEquipped ? _.without(equipped.weapons, id) : [...equipped.weapons, id]
			return setFieldValue('equipped', update(equipped, {
				weapons: { $set: weapons }
			}))
		} 
		return setFieldValue('equipped', update(equipped, {
			[category]: { $set: alreadyEquipped ? null : id }
		}))			
	}

	const SelectorContent = ({ currentItemId }) => {
		const currentItem = items[currentItemId]
		return (
			<div>
				<p><b>Category:</b> {currentItem.category}</p>

				{currentItem.weaponType && (
					<>
						<p><b>Type:</b> {currentItem.weaponType} / {currentItem.encumbrance} / {currentItem.damageType}</p>
						<p><b>Range:</b> {currentItem.range}</p>
						<p><b>Damage:</b> {currentItem.damage[_.lowerCase(race.size)]}</p>
						<p><b>Critical:</b> {currentItem.critical}</p>
					</>
				)}

				<p><b>Value:</b> {currentItem.value} gp</p>
				<p><b>Weight:</b> {currentItem.weight} lbs</p>
			</div>
		)
	}
	
	return (
		<div className="container--body">						
			<h3 className="section__title">Purchase Equipment</h3>

			<Selector
				items={availableItems} 
				Content={SelectorContent}
				handleSelect={handleSelect}
			/>

			<div className="divider"></div>

			<h4>Selected Equipment:</h4>

			<div className="section CreatorFormEquipment__grid">
				<h5 className="grid__col1">Item</h5>
				<h5 className="grid__col2">Qty</h5>
				<h5 className="grid__col3">Cost (gp)</h5>

				{_.map(equipment, item => {
					const id = item.id
					item = { id, ...items[id] }
					const qty = _.find(equipment, { id }).qty
					return (
						<Fragment key={`${id}Equipment`}>
							<button
								className="grid__col1 button--link"
								type="button"
								id={id}
								onClick={() => setClickedItem(item)}
							>
								{items[id].name}
							</button>

							<input 
								className="grid__col2 number-input"
								type="number"
								id={id}
								value={qty} 
								onChange={(e) => {
									const id = e.target.id
									const index = _.findIndex(equipment, { id })
									updateEquipment(update(equipment, {
										[index]: { $set: { id, qty: e.target.value } }
									}))
								}} 
							/>
							<div className="grid__col3">{calcItemTotalValue(item, qty)}</div>
							
							<EquipButton
								className="grid__col4"
								item={item} 
								equipped={equipped} 
								handleEquip={handleEquip} 
							/>
							
							<button 
								className="grid__col5 button"
								type="button" 
								onClick={() => setSelectedItemIds(_.without(selectedItemIds, id))}
							>
								<ion-icon name="close" size="small" />
							</button>
						</Fragment>
					)
				})}
			</div>
			<ErrorMessage className="CreatorForm__error" name="equipment" component="div" />
			
			<div className="CreatorForm__group">
				<div>Total weight: {totalWeight} lbs</div>
				<div>Remaining money: <span style={values.remainingGold < 0 ? { color: "red" } : null}>{values.remainingGold} gp</span></div>
			</div>
			<ErrorMessage className="CreatorForm__error--right" name="remainingGold" component="div" />

			<ItemModal
				characterSize={race.size}
				clickedItem={clickedItem}
				equipped={equipped}
				handleCloseModal={() => setClickedItem(undefined)}
				handleEquip={handleEquip}
			/>

		</div>
	)
}

export default CreatorFormEquipment