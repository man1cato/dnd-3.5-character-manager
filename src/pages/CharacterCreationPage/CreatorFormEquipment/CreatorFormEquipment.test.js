import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import _ from 'lodash'

import { apiData } from '../../../tests/utils'
import CreatorFormEquipment from './CreatorFormEquipment'


const setFieldValue = jest.fn()
let api, props

beforeAll(async () => {
   api = await apiData()
   props = {
      items: api.items,
      values: { 
         equipment: [],
         equipped: {
            armor: null,
            shield: null,
            weapons: []
         }
      },
      setFieldValue
   }
})


test('should correctly render CreatorFormEquipment', () => {
   const { container } = render(<CreatorFormEquipment {...props} />)
   expect(container.firstChild).toMatchSnapshot()
})