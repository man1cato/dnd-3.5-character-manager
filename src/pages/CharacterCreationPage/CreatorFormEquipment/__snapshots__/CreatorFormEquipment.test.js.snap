// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render CreatorFormEquipment 1`] = `
<div
  class="container--body"
>
  <h3
    class="section__title"
  >
    Purchase Equipment
  </h3>
  <div
    class="Selector__row"
  >
    <select
      class="Selector__dropdown select"
    >
      <option
        value="rec5S7j8Ef2sjxRQi"
      >
        Alchemist’s lab
      </option>
      <option
        value="recvNcEk3PNoStz2Z"
      >
        Arrow
      </option>
      <option
        value="recFXZvFLuLdLNUzT"
      >
        Artisan’s outfit
      </option>
      <option
        value="recSPxxuOkEV3f3pq"
      >
        Artisan’s tools
      </option>
      <option
        value="rec2enDkPMOblmSOa"
      >
        Artisan’s tools, masterwork
      </option>
      <option
        value="reczHJRu1GImUfB8d"
      >
        Backpack
      </option>
      <option
        value="recCOaodcn8UZNc5y"
      >
        Banded mail
      </option>
      <option
        value="recOvuth9QArkI54l"
      >
        Bastard Sword
      </option>
      <option
        value="recFAVnJfZ9TG5X29"
      >
        Battleaxe
      </option>
      <option
        value="rec35ieQLel7fW2Mo"
      >
        Bedroll
      </option>
      <option
        value="recwC6fYdv3dGEbmX"
      >
        Bite
      </option>
      <option
        value="recMfBuVg9RNlrAz8"
      >
        Block and tackle
      </option>
      <option
        value="recxH2pD5DK23RSPb"
      >
        Boar tusk
      </option>
      <option
        value="recgILJu3DU1KVSh8"
      >
        Bolas
      </option>
      <option
        value="recLA2TYeoO7FSYM7"
      >
        Bolt (Crossbow)
      </option>
      <option
        value="recSYqEwn9Uzr4mec"
      >
        Bolt (Repeating Crossbow)
      </option>
      <option
        value="rec3QuwY0QShwfije"
      >
        Breastplate
      </option>
      <option
        value="rec2zYSncfJJAb3iO"
      >
        Buckler
      </option>
      <option
        value="recYISSCN8m337WCk"
      >
        Bullet
      </option>
      <option
        value="recVn13EH7d6CttLq"
      >
        Caltrops
      </option>
      <option
        value="recle1X2VPvoL87q3"
      >
        Candle
      </option>
      <option
        value="recgbQQSOUc7KBaSB"
      >
        Chain (10 ft.)
      </option>
      <option
        value="recXSxRJI2r6eZ77y"
      >
        Chain shirt
      </option>
      <option
        value="recuMZnz4kks14Yku"
      >
        Chainmail
      </option>
      <option
        value="receFFMKpxA8xLcez"
      >
        Claws
      </option>
      <option
        value="recvadUwHBDtxxe1r"
      >
        Cleric’s vestments
      </option>
      <option
        value="recCV1j6Nm77BTPy7"
      >
        Climber’s kit
      </option>
      <option
        value="recq504GUQ9cVjYrZ"
      >
        Club
      </option>
      <option
        value="recaU4qOVoeOWwrla"
      >
        Cold weather outfit
      </option>
      <option
        value="recZ0cPuaD6obmPt6"
      >
        Composite  Shortbow
      </option>
      <option
        value="recrwMqSymePn4K1u"
      >
        Composite Longbow
      </option>
      <option
        value="recTxp2NBlNb2QC37"
      >
        Copper Piece
      </option>
      <option
        value="rec0nWhiOxO0qIoVl"
      >
        Courtier’s outfit
      </option>
      <option
        value="recbuPfJHmyjMHhQU"
      >
        Crowbar
      </option>
      <option
        value="reckinAA8j0zE1qYG"
      >
        Dagger
      </option>
      <option
        value="recQCWmEjsPEAoF0c"
      >
        Dart
      </option>
      <option
        value="recyQ6dGLHiby0auM"
      >
        Dire Flail
      </option>
      <option
        value="recQChWprYHDEq0s9"
      >
        Disguise kit
      </option>
      <option
        value="recdauUcanZYERAL0"
      >
        Displacer hide
      </option>
      <option
        value="recjSV9JrWX3jestd"
      >
        Displacer tentacle
      </option>
      <option
        value="reckMbImkYd1d2ZmH"
      >
        Dwarven Urgosh
      </option>
      <option
        value="recLjsCOMiz7Gp1Pg"
      >
        Dwarven Waraxe
      </option>
      <option
        value="recfj4dvzRdGIxINB"
      >
        Entertainer’s outfit
      </option>
      <option
        value="rec2u68zdokNfEcEx"
      >
        Explorer’s outfit
      </option>
      <option
        value="rec5rr8NvisUanp0q"
      >
        Falchion
      </option>
      <option
        value="recZBZ5DTJuo8Uaea"
      >
        Farsky's Elemental Shield
      </option>
      <option
        value="recVp9qZsA0E1MUr3"
      >
        Flail
      </option>
      <option
        value="rec1LVNOO5AqBGbRn"
      >
        Flint & steel
      </option>
      <option
        value="recgMQNb7xQdJc2Wo"
      >
        Full plate
      </option>
      <option
        value="recGlWG6qM0tD8JHx"
      >
        Gauntlet
      </option>
      <option
        value="reckaU5QIgjZbVC1y"
      >
        Glaive
      </option>
      <option
        value="recsazWObkJ7lKpfy"
      >
        Gnome Hooked Hammer
      </option>
      <option
        value="recdok7uDkyJPg6og"
      >
        Gold Piece
      </option>
      <option
        value="reclzkJslpRPe3BuH"
      >
        Gore
      </option>
      <option
        value="rec2oW1AeM9O03Qho"
      >
        Grappling hook
      </option>
      <option
        value="rec45mQ7GezQapeUK"
      >
        Greataxe
      </option>
      <option
        value="rectS2LFuBF2blTDs"
      >
        Greatclub
      </option>
      <option
        value="recOzNLJJONfYsYGz"
      >
        Greatsword
      </option>
      <option
        value="recEvB3sNR0ywy0eu"
      >
        Guisarme
      </option>
      <option
        value="recoEnfL2wwNnyS8v"
      >
        Halberd
      </option>
      <option
        value="recW8flYHrD72BAk7"
      >
        Half-plate
      </option>
      <option
        value="recaymuyizHxS6JHf"
      >
        Hammer
      </option>
      <option
        value="rec2OeTGGSrzGVMMG"
      >
        Hand Crossbow
      </option>
      <option
        value="reckrpTDGo1Z13XQu"
      >
        Handaxe
      </option>
      <option
        value="rec1sJBJfitBCoq16"
      >
        Healer’s kit
      </option>
      <option
        value="recKogDI571sIGJe8"
      >
        Heavy Crossbow
      </option>
      <option
        value="recYbCE2i7AyxzHJz"
      >
        Heavy Flail
      </option>
      <option
        value="recRrV57TGLtJi1dV"
      >
        Heavy Mace
      </option>
      <option
        value="recHD8oBfQaeBHb81"
      >
        Heavy Pick
      </option>
      <option
        value="recSadi8CZ8meVUe1"
      >
        Heavy Spiked Shield
      </option>
      <option
        value="recqJk1Xl657lul1U"
      >
        Heavy Wooden Shield
      </option>
      <option
        value="recz1UKhXvjxrG1cV"
      >
        Hide
      </option>
      <option
        value="recEiGPFtz5hiD7Oq"
      >
        Holly and mistletoe
      </option>
      <option
        value="recPA3EfImKRk28CF"
      >
        Holy symbol, silver
      </option>
      <option
        value="reclWOlqLHIhYkg1l"
      >
        Holy symbol, wooden
      </option>
      <option
        value="recOfWQd9ouMlEwu8"
      >
        Hourglass
      </option>
      <option
        value="recXzUW8Zh4kft34c"
      >
        Ivory tusk
      </option>
      <option
        value="recScg32IO6QTPViT"
      >
        Javelin
      </option>
      <option
        value="recLfMudQe4kgFhFH"
      >
        Kama
      </option>
      <option
        value="recMbCmpxjotB0VV7"
      >
        Kukri
      </option>
      <option
        value="recclEquKaEmbsVlx"
      >
        Lance
      </option>
      <option
        value="recqzAzbZksnBvBl3"
      >
        Leather
      </option>
      <option
        value="recWQ2ml4RhoLtvik"
      >
        Light Crossbow
      </option>
      <option
        value="recNpXB1tw0qzel1M"
      >
        Light Hammer
      </option>
      <option
        value="recNfkmUh5SEFUesV"
      >
        Light Mace
      </option>
      <option
        value="recmN3FqHCd5v6cOP"
      >
        Light Pick
      </option>
      <option
        value="rechNRpsAvDuJu3zu"
      >
        Light Shield
      </option>
      <option
        value="recbS5UzFUWtUqkPo"
      >
        Light Spiked Shield
      </option>
      <option
        value="receolEOHvI7Gxxs5"
      >
        Longbow
      </option>
      <option
        value="recNbhiub65qBJh76"
      >
        Longspear
      </option>
      <option
        value="recWrjoatwOuaSayt"
      >
        Longsword
      </option>
      <option
        value="recXPeop9fzPxp3mL"
      >
        Magnifying glass
      </option>
      <option
        value="recvQ9JtDMVLF6uTB"
      >
        Monk’s outfit
      </option>
      <option
        value="recLRqDjOPFzbQOjq"
      >
        Morningstar
      </option>
      <option
        value="recvACTRbCrHMjqvR"
      >
        Musical instrument, common
      </option>
      <option
        value="rec52IYY1rx6x44Nd"
      >
        Musical instrument, masterwork
      </option>
      <option
        value="recLCWZUrzv9vXDUa"
      >
        Net
      </option>
      <option
        value="recqV8ypVKKkf9cao"
      >
        Noble’s outfit
      </option>
      <option
        value="recPvSRFFsWx1LbP7"
      >
        Nunchaku
      </option>
      <option
        value="recSR6gCiAukUT92P"
      >
        Orc Double Axe
      </option>
      <option
        value="recCScbmMjzGHAyAB"
      >
        Padded
      </option>
      <option
        value="reclMGYunoAYSzwew"
      >
        Peasant’s outfit
      </option>
      <option
        value="recABkU6F0DFvkERT"
      >
        Platinum Piece
      </option>
      <option
        value="reclm7rkhfFy5fwH8"
      >
        Punching Dagger
      </option>
      <option
        value="rec0JezjbAvUiWr8A"
      >
        Quarterstaff
      </option>
      <option
        value="recTZljdQNmdT0tfE"
      >
        Ranseur
      </option>
      <option
        value="recLpRWzprhOXDwrU"
      >
        Rapier
      </option>
      <option
        value="rectukONY5k2HfJu2"
      >
        Rations
      </option>
      <option
        value="recCnIhAdgFLDiBK8"
      >
        Repeating Heavy Crossbow
      </option>
      <option
        value="recAaHr4qEcbytOpu"
      >
        Repeating Light Crossbow
      </option>
      <option
        value="recYxygXLa4JSCOrl"
      >
        Rope, hemp
      </option>
      <option
        value="rec5iII0mMA1Tr4DR"
      >
        Royal outfit
      </option>
      <option
        value="recParpRbpmkp9Wod"
      >
        Sai
      </option>
      <option
        value="recg3hmuoYeY6Vd22"
      >
        Sap
      </option>
      <option
        value="recCMz646OTItRZNF"
      >
        Scale mail
      </option>
      <option
        value="recvZibbog2OmcPuO"
      >
        Scale, merchant’s
      </option>
      <option
        value="recyIw9mRyuQleBdb"
      >
        Scholar’s outfit
      </option>
      <option
        value="recpb8WlBeCJbUryB"
      >
        Scimitar
      </option>
      <option
        value="recnrbjHjcF6Zw3nM"
      >
        Scythe
      </option>
      <option
        value="reczq2bvbYUKzykGa"
      >
        Shortbow
      </option>
      <option
        value="recKW7AEx35ZzDDgj"
      >
        Shortspear
      </option>
      <option
        value="recnIPKR672TnSnYl"
      >
        Shortsword
      </option>
      <option
        value="recaUDdSjNUO3z4X4"
      >
        Shovel
      </option>
      <option
        value="reccpQwnC44OsrGBE"
      >
        Shuriken
      </option>
      <option
        value="recXOLsAZLyI0PAeT"
      >
        Siangham
      </option>
      <option
        value="rec2q1Tae0EA9ppx1"
      >
        Sickle
      </option>
      <option
        value="recThYVmRyKPtBVTv"
      >
        Silk robe
      </option>
      <option
        value="rectXokKb6CNfiL9w"
      >
        Silver Piece
      </option>
      <option
        value="recgl5p6sMpgzwWo9"
      >
        Slap/Slam
      </option>
      <option
        value="reckm5B8oTdvUJjzI"
      >
        Sling
      </option>
      <option
        value="recqrC4ryQIqntqvi"
      >
        Spear
      </option>
      <option
        value="rec8HlxYbXDpOZcaK"
      >
        Spell component pouch
      </option>
      <option
        value="recl8vnhb1u28ROye"
      >
        Spiked Armor
      </option>
      <option
        value="recJB7Jom5jyJfqwg"
      >
        Spiked Chain
      </option>
      <option
        value="rec6MscBvV6I6fDsI"
      >
        Spiked Gauntlet
      </option>
      <option
        value="recqlsMONgIQXwWbS"
      >
        Splint mail
      </option>
      <option
        value="recmsrIsqwGO0q8Jr"
      >
        Spyglass
      </option>
      <option
        value="rectaWcLJh5xbEyBf"
      >
        Sting
      </option>
      <option
        value="receOmkviWX3nGzuV"
      >
        Studded leather
      </option>
      <option
        value="recDAXHjNpPpBuFsn"
      >
        Talons
      </option>
      <option
        value="recHYmG2tmeSsKInk"
      >
        Tentacle
      </option>
      <option
        value="rec4UM5rEuDdyMn7E"
      >
        Thieves’ tools
      </option>
      <option
        value="rechGBDSkQysibOAi"
      >
        Thieves’ tools, masterwork
      </option>
      <option
        value="recuOEI8Mv3a4wKSG"
      >
        Throwing Axe
      </option>
      <option
        value="recAhjHv1tXyRVdRp"
      >
        Tool, masterwork
      </option>
      <option
        value="recNMoaCq5Ng3DgkZ"
      >
        Torch
      </option>
      <option
        value="recvMMCsdSJEpZmcK"
      >
        Tower Shield
      </option>
      <option
        value="recofxRgpu8Sx62G5"
      >
        Traveler’s outfit
      </option>
      <option
        value="rec4gBxMabAvBRNEm"
      >
        Trident
      </option>
      <option
        value="recQVTxj40pWDtm8Y"
      >
        Two-Bladed Sword
      </option>
      <option
        value="recd96petEetM5NdO"
      >
        Unarmed Strike
      </option>
      <option
        value="rec6LbpVzZCv5DdR4"
      >
        Warhammer
      </option>
      <option
        value="recnOONApkyWGWJTW"
      >
        Water clock
      </option>
      <option
        value="recTnEbBAzGtBfdbJ"
      >
        Waterskin
      </option>
      <option
        value="recCg6CEUBzMwye5w"
      >
        Whip
      </option>
      <option
        value="reckwALiYfzmuSB3v"
      >
        Wizard's spellbook
      </option>
    </select>
    <button
      class="Selector__button button"
      type="button"
    >
      Select
    </button>
  </div>
  <div>
    <p>
      <b>
        Category:
      </b>
       
      Tool Kit
    </p>
    <p>
      <b>
        Value:
      </b>
       
      500
       gp
    </p>
    <p>
      <b>
        Weight:
      </b>
       
      40
       lbs
    </p>
  </div>
  <div
    class="divider"
  />
  <h4>
    Selected Equipment:
  </h4>
  <div
    class="section CreatorFormEquipment__grid"
  >
    <h5
      class="grid__col1"
    >
      Item
    </h5>
    <h5
      class="grid__col2"
    >
      Qty
    </h5>
    <h5
      class="grid__col3"
    >
      Cost (gp)
    </h5>
  </div>
  <div
    class="CreatorForm__group"
  >
    <div>
      Total weight: 
      0.00
       lbs
    </div>
    <div>
      Remaining money: 
      <span>
         gp
      </span>
    </div>
  </div>
</div>
`;
